AppSyncRealTimeClient/ConnectionProvider/AppsyncRealtimeConnection/RealtimeConnectionProvider+StaleConnection.swift

```diff
@@ -0,0 +1,49 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+extension RealtimeConnectionProvider {
+
+    /// Start a stale connection timer, first invalidating and destroying any existing timer
+    func startStaleConnectionTimer() {
+        AppSyncLogger.debug("Starting stale connection timer for \(staleConnectionTimeout)s")
+        if keepAliveTimer != nil {
+            stopStaleConnectionTimer()
+        }
+        keepAliveTimer = CountdownTimer(interval: staleConnectionTimeout) {
+            self.disconnectStaleConnection()
+        }
+    }
+
+    /// Stop and destroy any existing stale connection timer
+    func stopStaleConnectionTimer() {
+        AppSyncLogger.debug("Stopping and destroying stale connection timer")
+        keepAliveTimer?.invalidate()
+        keepAliveTimer = nil
```

Comment:
any reason to set the keepAliveTimer to `nil` instead of holding onto it and just restarting it?

Reply:
No reason other than to suggest its ephemerality. It's a lightweight construct that only exists for the purposes of tracking the current connection; this emphasizes in practice and semantically that there is a 1:1 mapping between a connection and its stale connection timer.

I've further emphasized that by renaming `keepAliveTimer` to `staleConnectionTimer`.